File: src/models/StorySuggestion.ts
import {Entity,PrimaryGeneratedColumn,Column,ManyToOne,JoinColumn} from 'typeorm';
import { Story } from './Story';

@Entity('story_suggestion')
export class StorySuggestion {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  story_id!: number;

  @Column({ type: 'varchar', nullable: true })
  about!: string;

  @Column({ type: 'text', nullable: true })
  suggestion!: string;

  @ManyToOne(() => Story, (story) => story.suggestions)
  @JoinColumn({ name: 'story_id' })
  story!: Story;
}

File: src/models/story_suggestion.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey
from sqlalchemy.orm import relationship
from src.db.database import Base


class StorySuggestion(Base):
    __tablename__ = "story_suggestion"

    id = Column(Integer, primary_key=True, index=True)
    story_id = Column(Integer, ForeignKey("story.id"), nullable=False)
    about = Column(String, nullable=True)
    suggestion = Column(Text, nullable=True)

    story = relationship("Story", back_populates="suggestions")
